### DISCUSSION TITLE
History syntax highlighting on zsh?

### DISCUSSION THREAD

**Discussion by Ioulan** [junegunn/fzf#3025] [Q&A] [answered]
Hello,
Is it possible to do this on zsh?
Similarly, like this on [fzf.fish](https://github.com/PatrickF1/fzf.fish) :
![](https://raw.githubusercontent.com/PatrickF1/fzf.fish/assets/history.png)

**Comment by yutkat** [comment, marked as answer]
If you want very simple highlighting, there is a method to use `bat`.

https://github.com/yutkat/dotfiles/blob/3e72b7af8d9acc4d87e995f48c5338d2c6cad422/.config/zsh/rc/myplugins/fzf-preview.sh#L23-L25

https://github.com/yutkat/dotfiles/blob/3e72b7af8d9acc4d87e995f48c5338d2c6cad422/.config/zsh/rc/pluginconfig/fzf_atload.zsh#L8

If you want to correctly highlight, there are these plugins, but I don't know how to color from stdout.
- https://github.com/zsh-users/zsh-syntax-highlighting
- https://github.com/zdharma-continuum/fast-syntax-highlighting

**Comment by Ioulan** [comment]
![image](https://user-images.githubusercontent.com/114978689/198228599-858aa988-a638-40b0-926a-42b7e2ae7986.png)
> If you want very simple highlighting, there is a method to use `bat`.

Yes, thank you very much, this one works (I've tried to use `bat` before, but It doesn't syntax highlight it, I was just missing this `--language=sh` flag.
> If you want to correctly highlight, there are these plugins, but I don't know how to color from stdout.
> - https://github.com/zsh-users/zsh-syntax-highlighting
> - https://github.com/zdharma-continuum/fast-syntax-highlighting

I'm also using this plugin [zsh-syntax-higlighting](https://github.com/zsh-users/zsh-syntax-highlighting), but it doesn't work for me.

Btw nice dotfiles and preview script (tried to use it though it didn't work for me xd)

**Reply by yutkat** [reply]
> I'm also using this plugin [zsh-syntax-higlighting](https://github.com/zsh-users/zsh-syntax-highlighting), but it doesn't work for me.

Yes. fzf display is not the normal prompt display, so it will not be colored.
You will have to hack that program yourself to accept standard input.

In addition, I gave it up because it was too difficult.

> Btw nice dotfiles and preview script (tried to use it though it didn't work for me xd)

:wink:
